photutils/psf/building/catalogs.py

```diff
@@ -0,0 +1,1537 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This module provides tools for source extraction from images and catalogs.
+"""
+from __future__ import absolute_import, division, print_function
+
+import warnings
+
+import numpy as np
+from astropy.table import Table
+from astropy.nddata import NDData
+from astropy.nddata.nddata import UnknownUncertainty
+from astropy.utils.exceptions import AstropyUserWarning
+from astropy import log, wcs
+
+from .centroid import find_peak
+from .utils import py2round, interpolate_missing_data
+
+__all__ = ['extract_stars', 'sim_extract_stars', 'Weights', 'Star',
+           'expand_starlist']
+
+
+class Weights(UnknownUncertainty):
+    """ Convenience class for defining weights to be used
+    with `~astropy.nddata.NDData` input images.
+
+    """
+    @property
+    def uncertainty_type(self):
+        return 'weights'
+
+
+class Star(object):
```

Comment:
If you insist, obviously you can derive it from almost anything. I am not sure though it will simplify anything. In addition, according to @eteq's wishes expressed in sprint meetings, ``Star`` should be replaced with astropy's tables.